stream-chat-android-client/src/main/java/io/getstream/chat/android/client/setup/state/ClientState.kt

```diff
@@ -71,22 +73,20 @@ public interface ClientState {
      */
     public val isInitialized: Boolean
 
+    public val internetAvailable: Boolean
+
     /**
      * Clears the state of [ClientState].
      */
     public fun clearState()
 
     public companion object {
 
-        private var instance: ClientState? = null
-
-        @InternalStreamChatApi
-        public fun get(): ClientState =
-            instance ?: create().also { clientState ->
-                instance = clientState
-            }
+        internal var instance: ClientState? = null
 
-        @InternalStreamChatApi
-        public fun create(): ClientState = ClientStateImpl()
+        public fun create(context: Context): ClientState =
+            ClientStateImpl(
+                NetworkStateProvider(context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager)
```

Comment:
We have a new implementation for `NetworkStateProvider` that might be more feasible for this use case - `NetworkLifecyclePublisher`. What do you think about reusing it here?